tests/unit/libbpf_test.cpp

```diff
@@ -626,6 +868,44 @@ TEST_CASE("libbpf map pinning", "[libbpf]")
     bpf_object__close(object);
 }
 
+TEST_CASE("libbpf obj pinning", "[libbpf]")
+{
+    _test_helper_libbpf test_helper;
+    const char* pin_path = "\\temp\\test";
+
+    struct bpf_object* object;
+    int program_fd;
+    int result = bpf_prog_load("droppacket.o", BPF_PROG_TYPE_XDP, &object, &program_fd);
+    REQUIRE(result == 0);
+    REQUIRE(object != nullptr);
+
+    struct bpf_map* map = bpf_map__next(nullptr, object);
+    REQUIRE(map != nullptr);
+
+    int map_fd = bpf_map__fd(map);
+    REQUIRE(map_fd > 0);
+
+    result = bpf_obj_pin(map_fd, pin_path);
+    REQUIRE(result == 0);
+
+    // No corresponding bpf_obj_unpin?
```

Comment:
FYI, there isn't one because on Linux the assumption is that one can use unlink(path) because everything is a pseudo-filesystem.   I mentioned this to some folks at LSF/MM/BPF where I think unpin apis need to be added to make libbpf more cross-platform friendly.